pkg/build/builder/dockerutil.go

```diff
@@ -28,7 +29,11 @@ type DockerClient interface {
 	RemoveImage(name string) error
 }
 
-// pushImage pushes a docker image to the registry specified in its tag
+// pushImage pushes a docker image to the registry specified in its tag.
+// The method will retry to push the image when following scenarios occur:
+// - Docker registry is down temporarily or permanently
```

Comment:
there is no way (from the returned error perspective) how to distinguish if the registry is down permanent or temporary.